### PR TITLE
[Docs] Venia Targets reference doc

### PR DESCRIPTION
**Pull request by jcalcaben**
## Description

This PR updates the JSDoc bocks in the `venia-ui-declare.js` file.
It also creates a file in devdocs that pulls this information into a reference-style topic.

## Related Issue

Closes PWA-647

## Acceptance 

@zetlen 

### Verification Stakeholders

@zetlen 

### Verification Steps

1. Navigate to the pwa devdocs directory: `cd pwa-devdocs`
1. Start the HTML preview server: `yarn develop`
1. Verify file is auto-generated: `src/_includes/auto-generated/venia-ui/lib/targets/venia-ui-declare.md`
1. On the devdocs preview site, in the top navigation, click on **Venia UI API** under the _Reference API_ item.
1. Verify the Extensibility Targets topic appears with content.
1. Verify this topic is listed on the left navigation for this section

## Checklist

- [ ] I have added tests to cover my changes, if necessary.
* I have updated the documentation accordingly, if necessary.

### CODE REVIEW

**Comment by devops-pwa-codebuild** [comment]
**Performance Test Results**

The following fails have been reported by WebpageTest. These numbers indicates a possible performance issue with the PR which requires further manual testing to validate.

https://pr-2472.pwa-venia.com : LH Performance Expected 0.85 Actual 0.52, LH Best Practices Expected 1 Actual 0.92
https://pr-2472.pwa-venia.com/venia-tops.html : LH Performance Expected 0.75 Actual 0.35, LH Best Practices Expected 1 Actual 0.92
https://pr-2472.pwa-venia.com/valeria-two-layer-tank.html : LH Performance Expected 0.8 Actual 0.5, LH Accessibility Expected 0.9 Actual 0.89, LH Best Practices Expected 1 Actual 0.92

File: packages/venia-ui/lib/targets/venia-ui-declare.js [hunk 1]
```diff
@@ -3,47 +3,33 @@
...
+         * @see [`richContentRenderers.tap()`]{@link richContentRenderers.tap}
+         * @see [Intercept function signature]{@link rendererInterceptFunction}
+         * @see [RichContentRenderers API]{@link RichContentRenderersAPI}
+         * @see [RichContentRenderer configuration]{@link RichContentRenderer}
          *
-         * @type {tapable.SyncHook}
-         * @param {rendererIntercept} callback
+         * @example <caption>Access the tapable object</caption>
+         * const veniaTargets = targets.of('@magento/venia-ui')
+         * const richContentRenderers = veniaTargets.richContentRenderers
```
[the remark this answers is not in the document]
Comment (jcalcaben) [reply packages/venia-ui/lib/targets/venia-ui-declare.js]: code sample updated

File: packages/venia-ui/lib/targets/venia-ui-declare.js [hunk 2]
```diff
@@ -3,93 +3,159 @@
...
-         */
-
-        /**
-         * Collects RichContentRenderers contributed by third party extensions.
-         * Builds an array of these renderers which Venia's RichContent
-         * component uses to try and render a block of "rich" content, most
-         * likely HTML.
+         * Use this target if your backend system uses a customized content
+         * storage format instead of plain HTML in "rich content" fields such
+         * as product descriptions and CMS blocks.
```
Comment (fooman) [inline packages/venia-ui/lib/targets/venia-ui-declare.js:20]: This is a slight inaccuracy as currently product descriptions use the RichText component for displaying descriptions
https://github.com/magento/pwa-studio/blob/develop/packages/venia-ui/lib/components/ProductFullDetail/productFullDetail.js#L167

Comment (zetlen) [reply packages/venia-ui/lib/targets/venia-ui-declare.js:20]: This is describing the fields in our API responses that may contain "rich content", not the frontend's ability to display them. It's true that our ProductFullDetail component doesn't use `<RichContent>` and we should fix that, but I think here we are talking about backend capabilities and not frontend.

**Pull request merged**